### DISCUSSION TITLE
is the reimplementation of light shader's shadow_vec possible?

### DISCUSSION THREAD

**Discussion by just-wasted** [godotengine/godot-proposals#8298] [2D]
In Godot 3.5 it is possible to manipulate shadows with the canvas_item shaders light{} builtin shadow_vec (inout). In Godot 4 this is no longer possible. Could this be reimplemented in the not so far future? I think its really useful to create good lighting on sprites which represent vertical/ elevated entities of the game world.

**Comment by AThousandShips** [comment]
How does this differ from the `SHADOW_VERTEX` in the `fragment` stage?

**Reply by just-wasted** [reply]
Since i'm new to shaders i don't know how this would differ. Playing around with `SHADOW_VERTEX` works well for sprites in top-down projection when setting `SHADOW_VERTEX.y.` to the distance from viewport-top to lower end of the sprite.
 
![grafik](https://github.com/godotengine/godot-proposals/assets/67445572/4f8e66b3-14cf-4c96-965f-eb59161a0858)


As far as i can tell from my limited experience, using this approach on isometric sprites seems not the way to go.

![grafik](https://github.com/godotengine/godot-proposals/assets/67445572/9ed44b09-3cef-44ea-823a-83fedafc5ed6)

In the Godot 2.1 isometric lighting demo `LIGHT_VEC` was used to correct shadow on walls. This didn't work on Godot 3 but `SHADOW_VEC` got introduced to help with this.

**Reply by AThousandShips** [reply]
Does the `LIGHT_VERTEX` work? Also have you tried utilizing the `z` component? Have you read the [documentation](https://docs.godotengine.org/en/stable/tutorials/shaders/shader_reference/canvas_item_shader.html#fragment-built-ins)?

**Reply by just-wasted** [reply]
`LIGHT_VERTEX` doesn't seem to have any affect on shadows. `z` in 2D is only relevant in combination with normal maps, as far as I know. 

I've read the documentation, but i have no clue how exactly `SHADOW_VERTEX` works and if it can be used for my use case. From my understanding it can be used to a) move the shadow map, b) set the shadow for x or y direction from a specified sample point in screen space, like i did in the first image. 
I've looked around the demo projects but didn't find one for `SHADOW_VERTEX` or `VERTEX`. Most fragment shaders I've found seem to only use `COLOR`.

**Comment by just-wasted** [comment]
update: https://github.com/godotengine/godot-proposals/issues/8339#issuecomment-1806918651
